fsspec/implementations/git.py

```diff
@@ -70,6 +70,9 @@ def _get_kwargs_from_urls(path):
         return out
 
     def ls(self, path, detail=True, ref=None, **kwargs):
+        def get_hex(obj):
```

Comment:
I suppose rather than redefine this on every call, it could be a top-level function in the module.

Reply:
I force-pushed to simply use `str(obj.id)` and tested it up until pygit2 1.4, which is the oldest version I was able to install. I suspect it works with all versions though.

Reply:
Based on the [changelog](https://github.com/libgit2/pygit2/blob/master/CHANGELOG.md#0202-2014-02-04) it seems to have been added in pygit2 0.20.2 in 2014.

> New Object.id deprecates Object.oid https://github.com/libgit2/pygit2/pull/322

Reply:
Quite recent, then :)